src/labels.js

```diff
@@ -0,0 +1,443 @@
+            labels[i] = null;
+          }
+        }
+        object.accounts.push(labels);
+      }
+    } else if (major > 1) {
+      // Payload contains unsuppored new major version, abort:
+      throw new Error('LABELS_UNSUPPORTED_MAJOR_VERSION');
+    } else if (major === 1 && minor > 0) {
+      // New minor version can safely be used in read-only mode:
+      this._readOnly = true;
+    } else if (major === 1 && minor === 0 && patch > 0) {
+      // New patch version can safely to be used.
+    }
+
+    // Run (future) migration scripts:
+    // switch (object.version) {
+    //   case '1.0.0':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.1';
+    //     // falls through
+    //   case '1.0.1':
+    //     // Migrate from 1.0.1 to e.g. 1.0.2 or 1.1.0:
+    //     // ...
+    //     object.version = '1.0.2';
+    //     // falls through
+    //   default:
+    // }
+    return object;
+  }
+
+  // Goes through all labeled addresses and checks which ones have transactions.
+  // This result will be cached in the future. Although we obtain the balance,
+  // this is an implementation detail and we don't save it.
+  checkIfUsed (accountIndex) {
+    assert(Helpers.isPositiveInteger(accountIndex), 'specify accountIndex');
+    let labeledAddresses = this.all(accountIndex).filter((a) => a !== null);
+    let addresses = labeledAddresses.map((a) => a.address);
+
+    if (addresses.length === 0) return Promise.resolve();
+
+    BlockchainAPI.getBalances(addresses).then((data) => {
+      for (let labeledAddress of labeledAddresses) {
+        if (data[labeledAddress.address]) {
+          labeledAddress.used = data[labeledAddress.address].n_tx > 0;
+        }
+      }
+    });
+
+    return Promise.resolve();
+  }
+
+  fetchBalance (hdAddresses) {
+    assert(Array.isArray(hdAddresses), 'hdAddresses missing');
+    for (let addressHD of hdAddresses) {
+      assert(addressHD.constructor.name === 'AddressHD', 'AddressHD required');
+    }
+    let addresses = hdAddresses.filter((a) => a.balance === null)
+               .map((a) => a.address);
+
+    if (addresses.length === 0) return Promise.resolve();
+
+    return BlockchainAPI.getBalances(addresses).then((data) => {
+      for (let a of hdAddresses) {
+        if (data[a.address]) {
+          a.used = data[a.address].n_tx > 0;
+          a.balance = data[a.address].final_balance;
+        }
+      }
+    });
+  }
+
+  all (accountIndex, options = {}) {
+    assert(Helpers.isPositiveInteger(accountIndex), 'specify accountIndex');
+    if (!this._accounts[accountIndex]) return null;
+
+    if (options.includeUnlabeled) {
+      // Add null entries up to the current receive index
+      let receiveIndex = this._wallet.hdwallet.accounts[accountIndex].receiveIndex;
+      for (let i = 0; i < receiveIndex; i++) {
+        if (!this._accounts[accountIndex][i]) {
+          this._accounts[accountIndex][i] = new AddressHD(null,
```

Comment:
It's a way to limit how many addresses are queried by `checkIfUsed`, but I'll see if I can refactor this.